examples_apps/run_tutorial_machine_learning.py

```diff
@@ -0,0 +1,319 @@
+print("Exact and TensorFlow gradient agree:", np.allclose(grad, 2*a))
+
+##############################################################################
+# Processing data
+# ---------------
+#
+# The parameters for Blackbird states, gates, and measurements may be more complex
+# than just raw data or machine learning weights. These can themselves be the outputs
+# from some learnable function like a neural network [#]_.
+#
+# We can also use the ``sf.math`` module to allow arbitrary processing of measurement
+# results within the Tensorflow backend, by manipulating the ``.par`` attribute of
+# the measured register. Note that the math functions in ``sf.math`` will be automatically
+# converted to the equivalent TensorFlow ``tf.math`` function:
+
+eng.reset()
+prog = sf.Program(2)
+
+with prog.context as q:
+    MeasureX | q[0]
+    Dgate(sf.math.sin(q[0].par)) | q[1]
+
+result = eng.run(prog)
+print("Measured Homodyne sample from mode 0:", result.samples[0][0])
+
+mean, var = result.state.mean_photon(0)
+print("Mean photon number of mode 0:", mean)
+
+mean, var = result.state.mean_photon(1)
+print("Mean photon number of mode 1:", mean)
+
+
+##############################################################################
+# The mean photon number of mode 1 should be given by :math:`\sin(q[0])^2`, where :math:`q[0]`
+# is the measured Homodyne sample value:
+
+q0 = result.samples[0][0]
+print(np.sin(q0)**2)
+
+##############################################################################
+# Working with batches
+# --------------------
+#
+# It is common in machine learning to process data in *batches*. Strawberry Fields supports both
+# unbatched and batched data when using the Tensorflow backend. Unbatched operation is the default
+# behaviour (shown above). To enable batched operation, you should provide an extra
+# ``batch_size`` argument [#]_ within the ``backend_options`` dictionary, e.g.,
+
+# run simulation in batched-processing mode
+batch_size = 3
+prog = sf.Program(1)
+eng = sf.Engine('tf', backend_options={"cutoff_dim": 15, "batch_size": batch_size})
+
+x = prog.params("x")
+
+with prog.context as q:
+    Thermal(x) | q[0]
+
+x_val = tf.Variable([0.1, 0.2, 0.3])
+result = eng.run(prog, args={"x": x_val})
+print("Mean photon number of mode 0 (batched):", result.state.mean_photon(0)[0])
+
+##############################################################################
+# .. note:: The batch size should be static, i.e., not changing over the course of a computation.
+#
+# Parameters supplied to a circuit in batch-mode operation can either be scalars or vectors
+# (of length ``batch_size``). Scalars are automatically broadcast over the batch dimension.
+
+alpha = tf.Variable([0.5] * batch_size)
+theta = tf.constant(0.0)
+phi = tf.Variable([0.1, 0.33, 0.5])
+
+##############################################################################
+# Measurement results will be returned as Tensors with shape ``(batch_size,)``.

```

Comment:
(Minor suggestion): or comparable to refer to the specific type of tensor
```suggestion
# Measurement results will be returned as ``tf.Tensor``s with shape ``(batch_size,)``.

```

Reply:
unfortunately this won't work, as final backticks must always be followed by spaces!